File: Kotlin/Factorial.kt
//Recursive Factorial coding interview question in Kotlin
//by Freya-The-Goddess

fun factorial(n: Int): Int {
    return if (n == 1) {
        n
    } else {
        n * factorial(n - 1) //recursive call
    }
}

fun main() {
    println(factorial(12))
}

File: JavaScript/Factorial.js
//Recursive Factorial coding interview question in JavaScript
//by Freya-The-Goddess

function factorial(n) {
    if (n == 1) {
        return n
    } else {
        return n * factorial(n - 1) //recursive call
    }
}

function main() {
    output = factorial(12).toString()
    document.getElementById("result").textContent = output;
}

